internal/mithril-persistence/src/database/repository/cardano_transaction_repository.rs

```diff
@@ -251,7 +268,11 @@ impl CardanoTransactionRepository {
         {
             let threshold = highest_block_range_start.saturating_sub(number_of_blocks_to_keep);
             let query = DeleteCardanoTransactionQuery::below_block_number_threshold(threshold)?;
-            self.connection.fetch_first(query)?;
+
+            let connection = self.connection_pool.connection()?;
+            let transaction = connection.begin_transaction()?;
```

Comment:
Is a transaction needed here, given that we execute only one SQL request?

Reply:
It is a `DELETE` transaction, so it should be included in a transaction to avoid locks on the table :+1:

Reply:
I think that @dlachaume is right:
> Any command that accesses the database (basically, any SQL command, except a few [PRAGMA](https://sqlite.org/pragma.html#syntax) statements) will automatically start a transaction if one is not already in effect. Automatically started transactions are committed when the last SQL statement finishes.

(cf 2. Transactions: https://sqlite.org/lang_transaction.html)